File: src/Rm.js
import { Command } from 'clipanion';

import Base from './Base';
import remove from './utils/remove';

export default class Rm extends Base {
  static paths = [['rm']];

  static usage = Command.Usage({
    description: 'Remove the not used containers',
    details: `
      This command would find the ids of the stopped container and remove them.
    `,
    examples: [[
      'Remove the stopped container',
      'ddocker rm',
    ]],
  });

  execute = () => remove(this.path[0], this.context);
}

File: src/rm.rs
use std::io::Error;

use clap::Command;

use crate::utils::sub_process;

pub fn command() -> Command<'static> {
    Command::new("rm")
        .about("Find the all ids of the stopped containers and remove them")
}

pub fn execute() -> Result<(), Error> {
    let stdout = sub_process::exec_result(
        "docker",
        vec!["ps", "-aq", "-f", "status=exited", "-f", "status=created"],
    )?;
    let ids: Vec<&str> = stdout.split("\n")
        .filter(|x| !x.is_empty())
        .collect();

    if ids.len() == 0 {
        println!("No containers need to be removed.");
        return Ok(());
    }

    sub_process::exec(
        "docker",
        [vec!["rm"], ids]
            .concat(),
    );
    Ok(())
}
